pkg/cmd/indices/delete/delete.go

```diff
@@ -116,3 +122,81 @@ func runDeleteCmd(opts *DeleteOptions) error {
 
 	return nil
 }
+
+// Delete a replica index.
+func deleteReplicaIndex(client *search.Client, replicaIndex *search.Index) error {
+	replicaName := replicaIndex.GetName()
+	primaryName, err := findPrimaryIndex(replicaIndex)
+	if err != nil {
+		return fmt.Errorf("can't find primary index for %q: %w", replicaName, err)
+	}
+
+	err = unlinkReplicaIndex(replicaName, primaryName, client)
+	if err != nil {
+		return fmt.Errorf("can't unlink replica index %s from primary index %s: %w", replicaName, primaryName, err)
+	}
+
+	_, err = replicaIndex.Delete()
+	if err != nil {
+		return fmt.Errorf("can't delete replica index %q: %w", replicaName, err)
+	}
+
+	return nil
+}
+
+// Find the primary index of a replica index
+func findPrimaryIndex(replicaIndex *search.Index) (string, error) {
+	replicaName := replicaIndex.GetName()
+	settings, err := replicaIndex.GetSettings()
+
+	if err != nil {
+		return "", fmt.Errorf("can't get settings of replica index %q: %w", replicaName, err)
+	}
+
+	primary := settings.Primary
+	if primary == nil {
+		return "", fmt.Errorf("index %s doesn't have a primary", replicaName)
+	}
+
+	return primary.Get(), nil
+}
+
+// Remove replica from `replicas` settings of the primary index
+func unlinkReplicaIndex(replicaName string, primaryName string, client *search.Client) error {
```

Comment:
What do you think about using the `detach` wording instead of `unlink`? I think this is the wording use in our documentation, like here: https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/in-depth/replicas/#deleting-or-detaching-a-replica